aptos-move/block-executor/src/executor.rs

```diff
@@ -279,6 +281,8 @@ where
                 Resource => versioned_cache.data().remove(&k, idx_to_execute),
                 Module => versioned_cache.modules().remove(&k, idx_to_execute),
                 Group => {
+                    needs_suffix_validation = true;
```

Comment:
I am missing something, why do we need this?

Reply:
anytime behavior that is observed during speculative execution (and this now includes group metadata and size) changes, we need to trigger suffix re-validation. Previously, we always wait on estimates, so the only way to change behavior is to write outside. Having a write that was there before but not anymore was not an issue - could only cause additional waiting but not an incorrect speculation result.

However, now, the size or metadata might be read, then speculative group update might be removed fully, and without triggering re-validation, some later transaction might end up with the incorrect read result (corresponding to the removed group information that only existed in some incorrect speculative state).

Funny enough, missing this case was exactly the bug I had before when it wasn't passing the tests :) Will add a comment because it is tricky.